### PR TITLE
203: Finite state machine pattern

### PR DESCRIPTION
**Pull request by Hrighter**
Pull request description

for issue: #203 
create a finite state machine pattern demo

### CODE REVIEW

File: finite-state-machine/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+---
+layout: pattern
+title: Finite state machine
```
Comment (iluwatar) [inline finite-state-machine/README.md:3]:
```suggestion
title: Finite State Machine
```

File: finite-state-machine/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+folder: Finite-state-machine
```
Comment (iluwatar) [inline finite-state-machine/README.md:4]:
```suggestion
folder: finite-state-machine
```

File: finite-state-machine/README.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+permalink: /patterns/Finite-state-machine/
```
Comment (iluwatar) [inline finite-state-machine/README.md:5]:
```suggestion
permalink: /patterns/finite-state-machine/
```

File: finite-state-machine/README.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+---
+
+## Intent
+It represents a mathematical model of a finite number of states and their transitions and actions, 
+and is an efficient programming method within a logical unit. Make the program logic clear and easy
+to understand.
+
+
+## Applicability
+Use the Finite-state-machine pattern when
```
Comment (iluwatar) [inline finite-state-machine/README.md:18]:
```suggestion
Use the Finite State Machine pattern when
```

File: finite-state-machine/src/main/java/com/iluwatar/finite/state/machine/App.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.iluwatar.finite.state.machine;
+
+/**.
```
Comment (iluwatar) [inline finite-state-machine/src/main/java/com/iluwatar/finite/state/machine/App.java:26]:
```suggestion
/**
```

File: finite-state-machine/src/main/java/com/iluwatar/finite/state/machine/App.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+ * This model named finite state machine model, which uses finite state machine to
+ * implement one thing that pull out all the states, events and actions, and manage
+ * the complex state migration logic uniformly.
+ * <p></p>
+ * In this demo, I create a scenario that there is a game character, who has three states.
+ * States: Standing;Walking;Running
+ * And also there are two events, which means you can create requests to control it.
+ * Events: MOVING_ON;MOVING_OFF
```
Comment (iluwatar) [inline finite-state-machine/src/main/java/com/iluwatar/finite/state/machine/App.java:34]: In the code however I see events such as MOVING_ON_OFF and MOVING_RUN?

File: finite-state-machine/src/main/java/com/iluwatar/finite/state/machine/RunState.java [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+      default:
+        break;
+    }
+  }
+
+  /**.
+   * A method which simulates Game character stop.
+   */
+  private void stop() {
+    System.out.println("Stop running! Game character is standing now!");
```
Comment (iluwatar) [inline finite-state-machine/src/main/java/com/iluwatar/finite/state/machine/RunState.java:55]: Please use logger for outputting. Check elsewhere too.

File: finite-state-machine/src/main/java/com/iluwatar/finite/state/machine/App.java (hunk 6 above)
Comment (iluwatar) [reply finite-state-machine/src/main/java/com/iluwatar/finite/state/machine/App.java:34]: Maybe these events are not the easiest to understand. How about a different system like increasing and decreasing movement? From standing to walking to running and vice versa.

**Comment by iluwatar** [comment]
@Hrighter please comment when you're ready for another review.

**Comment by iluwatar** [comment]
@Hrighter The pull request has remained inactive and is about to be closed. Please comment if you're still working on it.

**Comment by ohbus** [comment]
@Hrighter this PR has been inactive for a long time, we will close it soon if we do not see any updates.

**Comment by iluwatar** [comment]
Closed due to inactivity

**Pull request closed without merging**